<?xml version="1.0"?>
<!--
Software License Agreement (BSD)

\file      rover_teleop.launch
\authors   Wil Selby <accounts@wilselby.com>
\copyright Copyright (c) 2017, All rights reserved.

Redistribution and use in source and binary forms, with or without modification, are permitted provided that
the following conditions are met:
 * Redistributions of source code must retain the above copyright notice, this list of conditions and the
   following disclaimer.
 * Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
   following disclaimer in the documentation and/or other materials provided with the distribution.
 * The name of the contributors may not be used to endorse or promote products derived from this software 
   without specific prior written permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WAR-
RANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, IN-
DIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-->

<launch>

  <arg name="joy_dev" default="/dev/input/js0" />
  <arg name="joystick" default="true" />
  <arg name="max_linear_vel" default="2.0" />
  <arg name="max_angular_vel" default="0.34" />

  <group ns="joy_teleop">

  <!-- Load the Joystick node 
  <node respawn="true" pkg ="joy" type="joy_node" name="joy_node" >
    <param name="dev" type="string" value="/dev/input/js0" />
    <param name="deadzone" value="0.12" />
    <param name="autorepeat_rate" value="10.0" />
  </node>
  -->

  <!-- Load the Keyboard node -->
  <node pkg ="rover_teleop" type="turtlebot_teleop_key.py" name="key_teleop" output="screen" >
    <param name="forward_rate" value="0.3" />
    <param name="rotation_rate" value="2.5" />
    <remap from="key_vel" to="/key_teleop/cmd_vel" />
  </node>
  

  <!-- Load the Teleop node -->
  <node name="rover_joy_teleop" pkg="rover_teleop" type="rover_teleop" >
    <param name="axis_linear" value="1" type = "int" />
    <param name="axis_angular" value="0" type = "int" />
    <param name="max_linear_vel" value="$(arg max_linear_vel)" />
    <param name="max_angular_vel" value="$(arg max_angular_vel)" />
  </node>

  </group>

</launch>



